src/shared/Modal.tsx

```diff
@@ -0,0 +1,56 @@
+import { createSignal, Component, Show } from "solid-js";
+// Component Properties
+interface Props {
+  messageHeader: string;
+  message: string;
+  confirmText: string;
+  declineText: string;
+  onConfirm: () => void;
+}
+
+const Modal: Component<Props> = (props) => {
+  const [show, setShow] = createSignal(true);
+
+  const handleClose = () => {
+    setShow(false);
+  };
+  // Make sure you set an onConfirm function on the page or it will simply close the modal without performing a task!!
+  const handleConfirm = () => {
+    props.onConfirm();
+    setShow(false);
+  };
+
+  return (
+    <Show when={show()}>
+      <div class="fixed inset-x-0 bottom-0 px-4 pb-4 sm:inset-0 sm:flex sm:items-center sm:justify-center">
+        <div class="fixed inset-0 transition-opacity">
+          <div class="absolute inset-0 bg-gray-500 opacity-75" />
+        </div>
```

Comment:
Does this need to have a different z-index? On my machine, the background actually shows in front of the modal and stops me from interacting with it:

<img width="754" alt="Screen Shot 2023-02-18 at 18 54 40" src="https://user-images.githubusercontent.com/121980126/219901180-ceaa4dff-55a9-4608-b420-2a01f83bc7df.png">